src/test/java/org/minions/devfund/katerinaanzoleaga/battleship/ShipTest.java

```diff
@@ -0,0 +1,280 @@
+                assertFalse(lightCruiser.isSunk());
+            }
+        }
+        assertTrue(lightCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal cruiser.
+     */
+    @Test
+    public void testSunkCruiserH() {
+        final Ship cruiser = new Cruiser();
+        cruiser.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < cruiser.getLength(); ++i) {
+            cruiser.shootAt(0, i);
+            if (i != cruiser.getLength() - 1) {
+                assertFalse(cruiser.isSunk());
+            }
+        }
+        assertTrue(cruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical cruiser.
+     */
+    @Test
+    public void testSunkCruiserV() {
+        final Ship cruiser = new Cruiser();
+        assertFalse(cruiser.isSunk());
+        cruiser.placeShipAt(0, 0, false, theOcean);
+        for (int i = 0; i < cruiser.getLength(); ++i) {
+            cruiser.shootAt(i, 0);
+            if (i != cruiser.getLength() - 1) {
+                assertFalse(cruiser.isSunk());
+            }
+        }
+        assertTrue(cruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal battleCruiser.
+     */
+    @Test
+    public void testSunkBattleCruiserH() {
+        final Ship battleCruiser = new BattleCruiser();
+        battleCruiser.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < battleCruiser.getLength(); ++i) {
+            battleCruiser.shootAt(0, i);
+            if (i != battleCruiser.getLength() - 1) {
+                assertFalse(battleCruiser.isSunk());
+            }
+        }
+        assertTrue(battleCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical battleCruiser.
+     */
+    @Test
+    public void testSunkBattleCruiserV() {
+        final Ship battleCruiser = new BattleCruiser();
+        assertFalse(battleCruiser.isSunk());
+        battleCruiser.placeShipAt(0, 0, false, theOcean);
+        for (int i = 0; i < battleCruiser.getLength(); ++i) {
+            battleCruiser.shootAt(i, 0);
+            if (i != battleCruiser.getLength() - 1) {
+                assertFalse(battleCruiser.isSunk());
+            }
+        }
+        assertTrue(battleCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal battleShip.
+     */
+    @Test
+    public void testSunkBattleShipH() {
+        final Ship battleShip = new BattleShip();
+        battleShip.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < battleShip.getLength(); ++i) {
+            battleShip.shootAt(0, i);
+            if (i != battleShip.getLength() - 1) {
+                assertFalse(battleShip.isSunk());
+            }
+        }
+        assertTrue(battleShip.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical battleShip.
+     */
+    @Test
+    public void testSunkBattleShipCruiserV() {
+        final Ship battleShip = new BattleShip();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "battleShip" which hides the field declared at line 18. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)